test/eventsource_test.js

```diff
@@ -1268,3 +1268,31 @@ describe('Proxying', function () {
     })
   })
 })
+
+describe('REPORT method', function () {
+  it('declares support for setting method', function () {
+    assert.equal(true, EventSource.supportsSettingMethod)
+  })
+
+  it('performs REPORT request when method set to REPORT', function (done) {
```

Comment:
There's already a partial test for this: "can specify HTTP method and body" (line 545). It tests for POST, but this functionality isn't specific to REPORT anyway, it's for any method. The only thing that's missing there is that it's not checking the method. I would implement that by setting a variable outside of a closure, like that test is currently doing with `receivedContent` - simpler than putting a string in the response as you're doing here.

The test above this is fine, but the `describe` block should be called something more general like "EventSource object" since you're testing a general property of that; this isn't specific to the REPORT method.

Reply:
I think I missed the other test of method. It looks like it does actually check the method used on the request (line 552). Given that, do you think this test is superfluous? 

Agree on the describe block name, will update.

Reply:
Ah yeah, you're right - so this test is definitely redundant.